### ISSUE TITLE
Question about tokenized dataset

### ISSUE THREAD

**Issue by sudarshan85** [yang-zhang/lightning-language-modeling#1]
Hi,
Thank you for this repo. I'm working on building a language model for a medical dataset and this is really helpful.

I have one question about the tokenized datasets. Both the `line-by-line=True`  and `line-by-line=False` options result in similar datasets that don't form full sentences (or in case my dataset, full medical notes). My questions are:

1. What exactly are the difference between those options? I printed them out and I couldn't exactly tell the difference (other than more lines in one than the other)
2. Is it normal to have incomplete sentences as datapoints? Is that ok?
Thanks

**Comment by yang-zhang** [comment]
Hi 
For 1.
Please see
https://github.com/huggingface/transformers/blob/master/examples/language-modeling/run_mlm.py#L142-L145
    line_by_line: bool = field(
        default=False,
        metadata={"help": "Whether distinct lines of text in the dataset are to be handled as distinct sequences."},
    )
For 2. It is normal. Whether or not it's ok depends on the use case. If incomplete sentences are not missing important information then it's ok. For example, if you want to classify if a comment is toxic but the toxic part is at the tail of the sentence that's truncated, then it's difficult to predict.
Hope this helps and good luck.

**Comment by sudarshan85** [comment]
Thank you for your reply. That really helps! 

I guess I have to make a decision then. I am working with medical notes that have certain structure hand have line breaks all over the place. I went ahead and replaced the line breaks with a space to force the entire note to be read in as a single data point. It worked. However, I did get a warning that the number of tokens exceeds 512 (which is obvious). This would entail that the note will get truncated during training. Since, I'm hoping to build a fine-tuned model which can then be used as a pretrained model for downstream tasks, this might now be a big issue (I hope :) )
